### DISCUSSION TITLE
Problems connecting to wifi router

### DISCUSSION THREAD

**Discussion by obdevel** [earlephilhower/arduino-pico#2720] [Q&A]
I'm getting reports from my group about failures and extended delays in connecting to wifi routers. This aligns with my gut feel that connect times have increased, i.e. time to reach status WL_CONNECTED. What used to take 5 secs or so now regularly takes 15-20 secs. Nothing of significance has changed in my environment. Others have reported failures to connect at all or connections dropping after some time. I only ever saw this personally with an Alcatel pocket 4G router. Anecdotal evidence suggests this doesn't happen with micropython on the same hardware.

I tried a very simple sketch with debug turned on (Level: All) but I'm not seeing any debug output.

I also tried a sketch that attempts to reconnect after the router is rebooted and it regularly fails to reconnect.

Anything I can try to diagnose this ?

**Comment by earlephilhower** [comment]
We unfortunately have little insight into the WiFi stack proper.  It's all running in the proprietary blob on the 2ndary ARM chip running the WiFi radio.  I know from my ESP8266 Arduino Core maintainer days that IoT WiFi and home routers can sometimes be plain inscrutable.

The WiFi blob is part of the SDK (I believe it has to match the CYW43 driver rev).  You can move back to an earlier SDK version of the core (3.9.5 was last pre-2.0 SDK release) and see if the connection speeds change...
